src/pages/test-finalisation/pass-finalisation/pass-finalisation.ts

```diff
@@ -260,6 +265,70 @@ export class PassFinalisationPage extends PracticeableBasePageComponent {
     return ctrlHasErrors && this.form.get(this.passCertificateCtrl).dirty;
   }
 
+  // Custom validator for FormControls
+  validatePassCertificate(c: FormControl) {
+    return this.isPassCertificateValid(c.value) ? null :
+      {
+        validatePassCertificate: {
+          valid: false,
+        },
+      };
+  }
+
+  isPassCertificateValid(certificate: string | null): boolean {
+    if (!certificate || certificate.length !== 8) {
+      return false;
+    }
+    if (!this.isLetter(certificate[0])) {
+      return false;
+    }
+
+    const checkDigit = this.calculateMod37CheckDigit(certificate);
+
+    if (checkDigit === 'invalid') {
+      return false;
+    }
+
+    // compare entered checkdigit with generated one
+    if (checkDigit !== certificate.toUpperCase()[7]) {
```

Comment:
Are we certain that TARS won't care whether the checkDigit is in upper/lower case?

Reply:
we had a list of valid ones from submissions and it didn't care about case of check digit.